### PR TITLE
feat(react): update the design of the `Chip` component.

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose
Update the design of the `Chip` component.

Screenshot 2024-07-08 at 12 32 22

Screenshot 2024-07-08 at 12 32 58

Screenshot 2024-07-08 at 12 33 28

Screenshot 2024-07-08 at 14 04 14

Screenshot 2024-07-08 at 15 06 38

### Related Issues
- https://github.com/wso2/oxygen-ui/issues/221

### Checklist
- [ ] UX/UI review done on the final implementation.
- [ ] Story provided. (Add screenshots)
- [ ] Manual test round performed and verified.
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Documentation provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by pavinduLakshan — APPROVED** [review]

File: packages/react/src/components/Chip/chip.scss [hunk 1]
```diff
@@ -17,23 +17,99 @@
...
+    border-color: var(--oxygen-customComponents-Chip-properties-premium-border-color);
+    font-weight:  var(--oxygen-customComponents-Chip-properties-font-weight);
+    line-height: var(--oxygen-customComponents-Chip-properties-line-height);
+
+    .MuiChip-icon {
+      fill: var(--oxygen-customComponents-Chip-properties-premium-border-color);
+    }
+
+    .MuiChip-label {
+      -moz-background-clip: var(--oxygen-customComponents-Chip-properties--moz-background-clip);
```
Comment (brionmario) [inline packages/react/src/components/Chip/chip.scss:39]: We need to check if we are using `autoprefixer` with the `postcss` plugin.
If so, we can remove these vendor prefixes.

Comment (Achintha444) [reply packages/react/src/components/Chip/chip.scss]: fixed with https://github.com/wso2/oxygen-ui/pull/222/commits/0a43bdaf0351654be4b2ded8b5dd031b7190fe22

File: packages/react/src/components/Chip/chip.scss [hunk 2]
```diff
@@ -17,23 +17,90 @@
...
+  border-radius: var(--oxygen-customComponents-Chip-properties-border-radius);
+  border: var(--oxygen-customComponents-Chip-properties-border);
+  text-transform: var(--oxygen-customComponents-Chip-properties-text-transform);
+  font-weight: var(--oxygen-customComponents-Chip-properties-font-weight);
+  line-height: var(--oxygen-customComponents-Chip-properties-line-height);
+
   &-premium {
     background: var(--oxygen-customComponents-Chip-properties-premium-background);
-    color: var(--oxygen-customComponents-Chip-properties-premium-text-color);
+    border: var(--oxygen-customComponents-Chip-properties-border);
```
[the remark this answers is not in the document]
Comment (Achintha444) [reply packages/react/src/components/Chip/chip.scss:28]: true, but the issue is for these four chips we have 4 different class names,
- `oxygen-chip-premium`
- `oxygen-chip-beta`
- `oxygen-chip-new`
- `oxygen-chip-experimental`

these four classes are defined separately and need to add the styles separately, that's why the styles are repeated.

File: packages/react/src/components/Chip/chip.scss [hunk 3]
```diff
@@ -17,23 +17,90 @@
...
+    border-color: var(--oxygen-customComponents-Chip-properties-new-border-color);
+    font-weight:  var(--oxygen-customComponents-Chip-properties-font-weight);
+    line-height: var(--oxygen-customComponents-Chip-properties-line-height);
+
+    .MuiChip-icon {
+      fill: var(--oxygen-customComponents-Chip-properties-new-border-color);
+    }
+
+    .MuiChip-label {
+      background-clip: var(--oxygen-customComponents-Chip-properties-background-clip);
```
Comment (savindi7) [inline packages/react/src/components/Chip/chip.scss:60]: These repeated styles can be taken out..

**Review by brionmario — APPROVED** [review]

**Pull request merged**